In [1]:
# Importing package
import re
import pandas as pd
import numpy as np

# Load data

In [2]:
df = pd.read_csv("indicators_per-industry_per-county.csv")
df

Unnamed: 0   fips  county                    name industry_code  \
0                0   8001       1  Adams County, Colorado        113000   
1                1   8001       1  Adams County, Colorado        115000   
2                2   8001       1  Adams County, Colorado        211000   
3                3   8001       1  Adams County, Colorado        2122a0   
4                4   8001       1  Adams County, Colorado        212310   
...            ...    ...     ...                     ...           ...   
306027      306027  32510     510     Carson City, Nevada        812300   
306028      306028  32510     510     Carson City, Nevada        812900   
306029      306029  32510     510     Carson City, Nevada        813100   
306030      306030  32510     510     Carson City, Nevada        813a00   
306031      306031  32510     510     Carson City, Nevada        813b00   

                                          industry_detail  \
0                          timber and raw forest products   
1                        agriculture and forestry support   
2                                   unrefined oil and gas   
3                iron, gold, silver, and other metal ores   
4                                       dimensional stone   
...                                                   ...   
306027                           dry-cleaning and laundry   
306028  pet care, photofinishing, parking and other su...   
306029                            religious organizations   
306030  grantmaking, giving, and social advocacy organ...   
306031  civic, social, professional, and similar organ...   

               relevant_naics  year  payann  estab  ...          METL  MINE  \
0                       [113]  2012       0      1  ...  0.000000e+00   0.0   
1                       [115]  2012     290      6  ...  7.965438e-05   0.0   
2                       [211]  2012       0      6  ...  0.000000e+00   0.0   
3       [21221, 21222, 21229]  2012       0      1  ...  0.000000e+00   0.0   
4                     [21231]  2012       0      4  ...  0.000000e+00   0.0   
...                       ...   ...     ...    ...  ...           ...   ...   
306027                 [8123]  2012       0      4  ...  0.000000e+00   0.0   
306028                 [8129]  2012     331      4  ...  1.483303e-05   0.0   
306029                 [8131]  2012    2441     13  ...  1.670618e-07   0.0   
306030           [8132, 8133]  2012     635      9  ...  8.127071e-09   0.0   
306031           [8134, 8139]  2012    3487     27  ...  2.352356e-07   0.0   

              MSW  NREN          OZON          PEST       REN      SMOG  \
0        0.000000   0.0  0.000000e+00  0.000000e+00  0.000000  0.000000   
1        3.405586   0.0  1.827157e-04  8.769207e-03  0.958770  2.011564   
2        0.000000   0.0  0.000000e+00  0.000000e+00  0.000000  0.000000   
3        0.000000   0.0  0.000000e+00  0.000000e+00  0.000000  0.000000   
4        0.000000   0.0  0.000000e+00  0.000000e+00  0.000000  0.000000   
...           ...   ...           ...           ...       ...       ...   
306027   0.000000   0.0  0.000000e+00  0.000000e+00  0.000000  0.000000   
306028   2.148818   0.0  1.318396e-07  2.876990e-09  0.025598  0.076440   
306029   6.541864   0.0  0.000000e+00  0.000000e+00  0.000000  0.312826   
306030   4.713797   0.0  0.000000e+00  0.000000e+00  0.000000  0.020206   
306031  41.326242   0.0  2.745370e-09  0.000000e+00  0.000360  0.539562   

               VADD       WATR  
0          0.000000   0.000000  
1        211.423463  15.653502  
2          0.000000   0.000000  
3          0.000000   0.000000  
4          0.000000   0.000000  
...             ...        ...  
306027     0.000000   0.000000  
306028   191.834961   0.096584  
306029   696.409510   2.551873  
306030   440.264876   0.051385  
306031  2716.179671   5.801004  

[306032 rows x 35 columns]

# Data Processing

In [3]:
# Code is the first three digits of the industry code

df["code"] = df.industry_code.str.slice(0, 3)
df.head(20)

Unnamed: 0  fips  county                    name industry_code  \
0            0  8001       1  Adams County, Colorado        113000   
1            1  8001       1  Adams County, Colorado        115000   
2            2  8001       1  Adams County, Colorado        211000   
3            3  8001       1  Adams County, Colorado        2122a0   
4            4  8001       1  Adams County, Colorado        212310   
5            5  8001       1  Adams County, Colorado        2123a0   
6            6  8001       1  Adams County, Colorado        213111   
7            7  8001       1  Adams County, Colorado        21311a   
8            8  8001       1  Adams County, Colorado        221100   
9            9  8001       1  Adams County, Colorado        221300   
10          10  8001       1  Adams County, Colorado        311111   
11          11  8001       1  Adams County, Colorado        311119   
12          12  8001       1  Adams County, Colorado        311210   
13          13  8001       1  Adams County, Colorado        311225   
14          14  8001       1  Adams County, Colorado        311300   
15          15  8001       1  Adams County, Colorado        31151a   
16          16  8001       1  Adams County, Colorado        311615   
17          17  8001       1  Adams County, Colorado        31161a   
18          18  8001       1  Adams County, Colorado        311700   
19          19  8001       1  Adams County, Colorado        311810   

                                      industry_detail  \
0                      timber and raw forest products   
1                    agriculture and forestry support   
2                               unrefined oil and gas   
3            iron, gold, silver, and other metal ores   
4                                   dimensional stone   
5   sand, gravel, clay, phosphate, other nonmetall...   
6                                       well drilling   
7                 other support activities for mining   
8                                         electricity   
9             drinking water and wastewater treatment   
10                                   dog and cat food   
11                                  other animal food   
12                                   flours and malts   
13            refined vegetable, olive, and seed oils   
14                        sugar, candy, and chocolate   
15                              fluid milk and butter   
16                                   packaged poultry   
17                     packaged meat (except poultry)   
18                                            seafood   
19                        bread and other baked goods   

                      relevant_naics  year  payann  estab  ...          MINE  \
0                              [113]  2012       0      1  ...      0.000000   
1                              [115]  2012     290      6  ...      0.000000   
2                              [211]  2012       0      6  ...      0.000000   
3              [21221, 21222, 21229]  2012       0      1  ...      0.000000   
4                            [21231]  2012       0      4  ...      0.000000   
5                     [21232, 21239]  2012    2037      5  ...  14064.496597   
6                           [213111]  2012       0      2  ...      0.000000   
7   [213112, 213113, 213114, 213115]  2012       0     16  ...      0.000000   
8                             [2211]  2012       0     13  ...      0.000000   
9                             [2213]  2012    1833      6  ...      0.000000   
10                          [311111]  2012       0      1  ...      0.000000   
11                          [311119]  2012       0      3  ...      0.000000   
12                           [31121]  2012       0      1  ...      0.000000   
13                          [311225]  2012       0      1  ...      0.000000   
14                            [3113]  2012       0      2  ...      0.000000   
15                  [311511, 311512]  2012       0     

In [4]:
df.columns

Index(['Unnamed: 0', 'fips', 'county', 'name', 'industry_code',
       'industry_detail', 'relevant_naics', 'year', 'payann', 'estab', 'emp',
       'ACID', 'ENRG', 'ETOX', 'EUTR', 'FOOD', 'GCC', 'HAPS', 'HAZW', 'HC',
       'HNC', 'HRSP', 'HTOX', 'JOBS', 'LAND', 'METL', 'MINE', 'MSW', 'NREN',
       'OZON', 'PEST', 'REN', 'SMOG', 'VADD', 'WATR', 'code'],
      dtype='object')

In [5]:
# List of 3-digit industry codes (70 codes)

codes = df.code.unique()
codes

array(['113', '115', '211', '212', '213', '221', '311', '314', '315',
       '316', '321', '322', '323', '324', '325', '326', '327', '331',
       '332', '333', '334', '335', '336', '337', '339', '441', '445',
       '452', '481', '484', '485', '486', '48a', '492', '493', '511',
       '512', '515', '517', '518', '519', '522', '523', '524', '52a',
       '531', '532', '533', '541', '550', '561', '562', '611', '621',
       '622', '623', '624', '711', '713', '721', '722', '811', '812',
       '813', '712', '312', '313', '525', '114', '483'], dtype=object)

In [6]:
df_new = df.groupby(['fips', 'name', 'year', 'code'], as_index=False)['payann', 'estab', 'emp',
    'ACID', 'ENRG', 'ETOX', 'EUTR', 'FOOD', 'GCC', 'HAPS', 'HAZW', 'HC', 
    'HNC', 'HRSP', 'HTOX', 'JOBS', 'LAND', 'METL', 'MINE', 'MSW', 'NREN',
       'OZON', 'PEST', 'REN', 'SMOG', 'VADD', 'WATR'].sum()
df_new

fips                     name  year code  payann  estab  emp  \
0        1001  Autauga County, Alabama  2012  113    1102      6   30   
1        1001  Autauga County, Alabama  2012  212       0      2    0   
2        1001  Autauga County, Alabama  2012  221   16657      9  182   
3        1001  Autauga County, Alabama  2012  311       0      2    0   
4        1001  Autauga County, Alabama  2012  313       0      1    0   
...       ...                      ...   ...  ...     ...    ...  ...   
142910  56045   Weston County, Wyoming  2012  721       0      6    0   
142911  56045   Weston County, Wyoming  2012  722    1438     17   28   
142912  56045   Weston County, Wyoming  2012  811     585      8   24   
142913  56045   Weston County, Wyoming  2012  812       0      2    0   
142914  56045   Weston County, Wyoming  2012  813     177     11   15   

              ACID           ENRG        ETOX  ...          METL  MINE  \
0         0.861581       0.275217  893.717406  ...  6.592757e-05   0.0   
1         0.000000       0.000000    0.000000  ...  0.000000e+00   0.0   
2       284.165607  728617.651665  204.647213  ...  4.711330e-02   0.0   
3         0.000000       0.000000    0.000000  ...  0.000000e+00   0.0   
4         0.000000       0.000000    0.000000  ...  0.000000e+00   0.0   
...            ...            ...         ...  ...           ...   ...   
142910    0.000000       0.000000    0.000000  ...  0.000000e+00   0.0   
142911    0.006143       0.043024    0.260775  ...  1.198482e-05   0.0   
142912    0.001840       0.060362    0.010141  ...  6.116848e-06   0.0   
142913    0.000000       0.000000    0.000000  ...  0.000000e+00   0.0   
142914    0.000543       0.000000    0.000506  ...  1.211386e-08   0.0   

              MSW           NREN          OZON          PEST            REN  \
0        3.210888       0.000000  3.767853e-07  1.218061e-02       0.275217   
1        0.000000       0.000000  0.000000e+00  0.000000e+00       0.000000   
2       23.518978  435169.166702  1.886537e-03  5.996926e-06  293448.484963   
3        0.000000       0.000000  0.000000e+00  0.000000e+00       0.000000   
4        0.000000       0.000000  0.000000e+00  0.000000e+00       0.000000   
...           ...            ...           ...           ...            ...   
142910   0.000000       0.000000  0.000000e+00  0.000000e+00       0.000000   
142911  57.259023       0.000000  1.579423e-06  1.823160e-09       0.043024   
142912   5.895470       0.000000  1.133122e-05  2.051391e-08       0.060362   
142913   0.000000       0.000000  0.000000e+00  0.000000e+00       0.000000   
142914   0.474359       0.000000  0.000000e+00  0.000000e+00       0.000000   

               SMOG          VADD         WATR  
0         20.041190    654.873075   136.234634  
1          0.000000      0.000000     0.000000  
2       2400.143851  11995.153334  7473.552102  
3          0.000000      0.000000     0.000000  
4          0.000000      0.000000     0.000000  
...             ...           ...          ...  
142910     0.000000      0.000000     0.000000  
142911     0.686682    756.297604     2.082064  
142912     1.615604    361.835769     0.317093  
142913     0.000000      0.000000     0.000000  
142914     0.022683     50.497535     0.185040  

[142915 rows x 31 columns]

In [7]:
df_new.to_csv("data_US.csv", index=False)

Now an issue with above dataset is that if we look at a given county, it does not cover all the industries. 
So what we will do below is that we will fill null values for industries in the county that are not covered in above dataset.

# Second table with dummy values

In [8]:
df1 = df[['fips', 'name', 'year']].drop_duplicates().sort_values(by=['fips']).reset_index(drop=True)
df1['code'] = [codes.tolist()] * len(df1)
df1

fips                        name  year  \
0      1001     Autauga County, Alabama  2012   
1      1003     Baldwin County, Alabama  2012   
2      1005     Barbour County, Alabama  2012   
3      1007        Bibb County, Alabama  2012   
4      1009      Blount County, Alabama  2012   
...     ...                         ...   ...   
3135  56037  Sweetwater County, Wyoming  2012   
3136  56039       Teton County, Wyoming  2012   
3137  56041       Uinta County, Wyoming  2012   
3138  56043    Washakie County, Wyoming  2012   
3139  56045      Weston County, Wyoming  2012   

                                                   code  
0     [113, 115, 211, 212, 213, 221, 311, 314, 315, ...  
1     [113, 115, 211, 212, 213, 221, 311, 314, 315, ...  
2     [113, 115, 211, 212, 213, 221, 311, 314, 315, ...  
3     [113, 115, 211, 212, 213, 221, 311, 314, 315, ...  
4     [113, 115, 211, 212, 213, 221, 311, 314, 315, ...  
...                                                 ...  
3135  [113, 115, 211, 212, 213, 221, 311, 314, 315, ...  
3136  [113, 115, 211, 212, 213, 221, 311, 314, 315, ...  
3137  [113, 115, 211, 212, 213, 221, 311, 314, 315, ...  
3138  [113, 115, 211, 212, 213, 221, 311, 314, 315, ...  
3139  [113, 115, 211, 212, 213, 221, 311, 314, 315, ...  

[3140 rows x 4 columns]

In [9]:
df1 = df1.explode('code')
df1

fips                     name  year code
0      1001  Autauga County, Alabama  2012  113
0      1001  Autauga County, Alabama  2012  115
0      1001  Autauga County, Alabama  2012  211
0      1001  Autauga County, Alabama  2012  212
0      1001  Autauga County, Alabama  2012  213
...     ...                      ...   ...  ...
3139  56045   Weston County, Wyoming  2012  312
3139  56045   Weston County, Wyoming  2012  313
3139  56045   Weston County, Wyoming  2012  525
3139  56045   Weston County, Wyoming  2012  114
3139  56045   Weston County, Wyoming  2012  483

[219800 rows x 4 columns]

So now the above DataFrame has 219800 rows, which is 3140 counties * 70 industry codes!

In [10]:
df1.to_csv("unique.csv", index=False)

In [11]:
# Filling with null values
vals = ['payann', 'estab', 'emp','ACID', 'ENRG', 'ETOX', 'EUTR', 'FOOD', 'GCC', 'HAPS', 'HAZW', 'HC', 
    'HNC', 'HRSP', 'HTOX', 'JOBS', 'LAND', 'METL', 'MINE', 'MSW', 'NREN',
       'OZON', 'PEST', 'REN', 'SMOG', 'VADD', 'WATR']
df1[vals] = pd.DataFrame([np.zeros(len(vals))], index=df1.index)
df1

fips                     name  year code  payann  estab  emp  ACID  \
0      1001  Autauga County, Alabama  2012  113     0.0    0.0  0.0   0.0   
0      1001  Autauga County, Alabama  2012  115     0.0    0.0  0.0   0.0   
0      1001  Autauga County, Alabama  2012  211     0.0    0.0  0.0   0.0   
0      1001  Autauga County, Alabama  2012  212     0.0    0.0  0.0   0.0   
0      1001  Autauga County, Alabama  2012  213     0.0    0.0  0.0   0.0   
...     ...                      ...   ...  ...     ...    ...  ...   ...   
3139  56045   Weston County, Wyoming  2012  312     0.0    0.0  0.0   0.0   
3139  56045   Weston County, Wyoming  2012  313     0.0    0.0  0.0   0.0   
3139  56045   Weston County, Wyoming  2012  525     0.0    0.0  0.0   0.0   
3139  56045   Weston County, Wyoming  2012  114     0.0    0.0  0.0   0.0   
3139  56045   Weston County, Wyoming  2012  483     0.0    0.0  0.0   0.0   

      ENRG  ETOX  ...  METL  MINE  MSW  NREN  OZON  PEST  REN  SMOG  VADD  \
0      0.0   0.0  ...   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   
0      0.0   0.0  ...   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   
0      0.0   0.0  ...   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   
0      0.0   0.0  ...   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   
0      0.0   0.0  ...   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   
...    ...   ...  ...   ...   ...  ...   ...   ...   ...  ...   ...   ...   
3139   0.0   0.0  ...   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   
3139   0.0   0.0  ...   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   
3139   0.0   0.0  ...   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   
3139   0.0   0.0  ...   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   
3139   0.0   0.0  ...   0.0   0.0  0.0   0.0   0.0   0.0  0.0   0.0   0.0   

      WATR  
0      0.0  
0      0.0  
0      0.0  
0      0.0  
0      0.0  
...    ...  
3139   0.0  
3139   0.0  
3139   0.0  
3139   0.0  
3139   0.0  

[219800 rows x 31 columns]

Now, we concatenate the two tables.

In [12]:
result = pd.concat([df_new, df1], sort=False)
result

fips                     name  year code   payann  estab    emp  \
0      1001  Autauga County, Alabama  2012  113   1102.0    6.0   30.0   
1      1001  Autauga County, Alabama  2012  212      0.0    2.0    0.0   
2      1001  Autauga County, Alabama  2012  221  16657.0    9.0  182.0   
3      1001  Autauga County, Alabama  2012  311      0.0    2.0    0.0   
4      1001  Autauga County, Alabama  2012  313      0.0    1.0    0.0   
...     ...                      ...   ...  ...      ...    ...    ...   
3139  56045   Weston County, Wyoming  2012  312      0.0    0.0    0.0   
3139  56045   Weston County, Wyoming  2012  313      0.0    0.0    0.0   
3139  56045   Weston County, Wyoming  2012  525      0.0    0.0    0.0   
3139  56045   Weston County, Wyoming  2012  114      0.0    0.0    0.0   
3139  56045   Weston County, Wyoming  2012  483      0.0    0.0    0.0   

            ACID           ENRG        ETOX  ...      METL  MINE        MSW  \
0       0.861581       0.275217  893.717406  ...  0.000066   0.0   3.210888   
1       0.000000       0.000000    0.000000  ...  0.000000   0.0   0.000000   
2     284.165607  728617.651665  204.647213  ...  0.047113   0.0  23.518978   
3       0.000000       0.000000    0.000000  ...  0.000000   0.0   0.000000   
4       0.000000       0.000000    0.000000  ...  0.000000   0.0   0.000000   
...          ...            ...         ...  ...       ...   ...        ...   
3139    0.000000       0.000000    0.000000  ...  0.000000   0.0   0.000000   
3139    0.000000       0.000000    0.000000  ...  0.000000   0.0   0.000000   
3139    0.000000       0.000000    0.000000  ...  0.000000   0.0   0.000000   
3139    0.000000       0.000000    0.000000  ...  0.000000   0.0   0.000000   
3139    0.000000       0.000000    0.000000  ...  0.000000   0.0   0.000000   

               NREN          OZON      PEST            REN         SMOG  \
0          0.000000  3.767853e-07  0.012181       0.275217    20.041190   
1          0.000000  0.000000e+00  0.000000       0.000000     0.000000   
2     435169.166702  1.886537e-03  0.000006  293448.484963  2400.143851   
3          0.000000  0.000000e+00  0.000000       0.000000     0.000000   
4          0.000000  0.000000e+00  0.000000       0.000000     0.000000   
...             ...           ...       ...            ...          ...   
3139       0.000000  0.000000e+00  0.000000       0.000000     0.000000   
3139       0.000000  0.000000e+00  0.000000       0.000000     0.000000   
3139       0.000000  0.000000e+00  0.000000       0.000000     0.000000   
3139       0.000000  0.000000e+00  0.000000       0.000000     0.000000   
3139       0.000000  0.000000e+00  0.000000       0.000000     0.000000   

              VADD         WATR  
0       654.873075   136.234634  
1         0.000000     0.000000  
2     11995.153334  7473.552102  
3         0.000000     0.000000  
4         0.000000     0.000000  
...            ...          ...  
3139      0.000000     0.000000  
3139      0.000000     0.000000  
3139      0.000000     0.000000  
3139      0.000000     0.000000  
3139      0.000000     0.000000  

[362715 rows x 31 columns]

In [13]:
df_final = result.groupby(['fips', 'name', 'year', 'code'], as_index=False)['payann', 'estab', 'emp',
    'ACID', 'ENRG', 'ETOX', 'EUTR', 'FOOD', 'GCC', 'HAPS', 'HAZW', 'HC', 
    'HNC', 'HRSP', 'HTOX', 'JOBS', 'LAND', 'METL', 'MINE', 'MSW', 'NREN',
       'OZON', 'PEST', 'REN', 'SMOG', 'VADD', 'WATR'].sum()
df_final.head(62)

fips                     name  year code   payann  estab    emp      ACID  \
0   1001  Autauga County, Alabama  2012  113   1102.0    6.0   30.0  0.861581   
1   1001  Autauga County, Alabama  2012  114      0.0    0.0    0.0  0.000000   
2   1001  Autauga County, Alabama  2012  115      0.0    0.0    0.0  0.000000   
3   1001  Autauga County, Alabama  2012  211      0.0    0.0    0.0  0.000000   
4   1001  Autauga County, Alabama  2012  212      0.0    2.0    0.0  0.000000   
..   ...                      ...   ...  ...      ...    ...    ...       ...   
57  1001  Autauga County, Alabama  2012  611      0.0    3.0    0.0  0.000000   
58  1001  Autauga County, Alabama  2012  621  22257.0   66.0  554.0  0.030796   
59  1001  Autauga County, Alabama  2012  622      0.0    1.0    0.0  0.000000   
60  1001  Autauga County, Alabama  2012  623      0.0    7.0    0.0  0.000000   
61  1001  Autauga County, Alabama  2012  624   4718.0   23.0  287.0  0.011248   

        ENRG        ETOX  ...      METL  MINE         MSW  NREN          OZON  \
0   0.275217  893.717406  ...  0.000066   0.0    3.210888   0.0  3.767853e-07   
1   0.000000    0.000000  ...  0.000000   0.0    0.000000   0.0  0.000000e+00   
2   0.000000    0.000000  ...  0.000000   0.0    0.000000   0.0  0.000000e+00   
3   0.000000    0.000000  ...  0.000000   0.0    0.000000   0.0  0.000000e+00   
4   0.000000    0.000000  ...  0.000000   0.0    0.000000   0.0  0.000000e+00   
..       ...         ...  ...       ...   ...         ...   ...           ...   
57  0.000000    0.000000  ...  0.000000   0.0    0.000000   0.0  0.000000e+00   
58  0.712892    0.061228  ...  0.000655   0.0  103.482286   0.0  3.057976e-07   
59  0.000000    0.000000  ...  0.000000   0.0    0.000000   0.0  0.000000e+00   
60  0.000000    0.000000  ...  0.000000   0.0    0.000000   0.0  0.000000e+00   
61  0.120905    0.016761  ...  0.000113   0.0  105.879528   0.0  3.555660e-08   

            PEST       REN       SMOG          VADD        WATR  
0   1.218061e-02  0.275217  20.041190    654.873075  136.234634  
1   0.000000e+00  0.000000   0.000000      0.000000    0.000000  
2   0.000000e+00  0.000000   0.000000      0.000000    0.000000  
3   0.000000e+00  0.000000   0.000000      0.000000    0.000000  
4   0.000000e+00  0.000000   0.000000      0.000000    0.000000  
..           ...       ...        ...           ...         ...  
57  0.000000e+00  0.000000   0.000000      0.000000    0.000000  
58  1.274279e-07  0.712892   1.189888  14383.609633    5.913537  
59  0.000000e+00  0.000000   0.000000      0.000000    0.000000  
60  0.000000e+00  0.000000   0.000000      0.000000    0.000000  
61  2.201518e-08  0.120905   0.454761   3195.904517    2.805043  

[62 rows x 31 columns]

In [14]:
# Keeping the social factors here, because we will ignore it for future pivot operation ..
social_factors = df_final.iloc[:,0:7]
social_factors

fips                     name  year code  payann  estab   emp
0        1001  Autauga County, Alabama  2012  113  1102.0    6.0  30.0
1        1001  Autauga County, Alabama  2012  114     0.0    0.0   0.0
2        1001  Autauga County, Alabama  2012  115     0.0    0.0   0.0
3        1001  Autauga County, Alabama  2012  211     0.0    0.0   0.0
4        1001  Autauga County, Alabama  2012  212     0.0    2.0   0.0
...       ...                      ...   ...  ...     ...    ...   ...
219795  56045   Weston County, Wyoming  2012  721     0.0    6.0   0.0
219796  56045   Weston County, Wyoming  2012  722  1438.0   17.0  28.0
219797  56045   Weston County, Wyoming  2012  811   585.0    8.0  24.0
219798  56045   Weston County, Wyoming  2012  812     0.0    2.0   0.0
219799  56045   Weston County, Wyoming  2012  813   177.0   11.0  15.0

[219800 rows x 7 columns]

# Pivoting the table

In [15]:
table = pd.pivot_table(df, values=['ACID', 'ENRG', 'ETOX', 'EUTR', 'FOOD', 'GCC', 'HAPS', 'HAZW', 'HC', 
    'HNC', 'HRSP', 'HTOX', 'JOBS', 'LAND', 'METL', 'MINE', 'MSW', 'NREN',
       'OZON', 'PEST', 'REN', 'SMOG', 'VADD', 'WATR'], index=['fips', 'name', 'year'],
                    columns=['code'], aggfunc=np.sum).fillna(0)

In [16]:
table

ACID                           \
code                                        113  114       115       211   
fips  name                       year                                      
1001  Autauga County, Alabama    2012  0.861581  0.0  0.000000  0.000000   
1003  Baldwin County, Alabama    2012  0.533211  0.0  0.092654  0.000000   
1005  Barbour County, Alabama    2012  3.092154  0.0  0.000000  0.000000   
1007  Bibb County, Alabama       2012  1.713780  0.0  0.000000  0.000000   
1009  Blount County, Alabama     2012  0.283806  0.0  0.000000  0.000000   
...                                         ...  ...       ...       ...   
56037 Sweetwater County, Wyoming 2012  0.000000  0.0  0.172337  0.000000   
56039 Teton County, Wyoming      2012  0.000000  0.0  0.000000  0.000000   
56041 Uinta County, Wyoming      2012  0.000000  0.0  0.000000  0.000000   
56043 Washakie County, Wyoming   2012  0.000000  0.0  0.000000  5.156900   
56045 Weston County, Wyoming     2012  0.000000  0.0  0.000000  5.298274   

                                                                          \
code                                          212        213         221   
fips  name                       year                                      
1001  Autauga County, Alabama    2012    0.000000   0.000000  284.165607   
1003  Baldwin County, Alabama    2012    2.894175   0.000000    2.034845   
1005  Barbour County, Alabama    2012    0.000000   0.000000    0.000000   
1007  Bibb County, Alabama       2012    0.000000   0.000000    0.000000   
1009  Blount County, Alabama     2012    0.000000   0.000000    0.000000   
...                                           ...        ...         ...   
56037 Sweetwater County, Wyoming 2012  138.220686  52.325460    0.000000   
56039 Teton County, Wyoming      2012    0.000000   0.000000    0.000000   
56041 Uinta County, Wyoming      2012    0.000000  27.740218    0.000000   
56043 Washakie County, Wyoming   2012    0.000000   0.000000    0.000000   
56045 Weston County, Wyoming     2012    0.000000   2.373792    0.000000   

                                                           ...       WATR  \
code                                        311  312  313  ...        623   
fips  name                       year                      ...              
1001  Autauga County, Alabama    2012  0.000000  0.0  0.0  ...   0.000000   
1003  Baldwin County, Alabama    2012  0.005378  0.0  0.0  ...  22.062101   
1005  Barbour County, Alabama    2012  0.000000  0.0  0.0  ...   0.000000   
1007  Bibb County, Alabama       2012  0.000000  0.0  0.0  ...   0.000000   
1009  Blount County, Alabama     2012  0.000000  0.0  0.0  ...   0.715589   
...                                         ...  ...  ...  ...        ...   
56037 Sweetwater County, Wyoming 2012  0.000000  0.0  0.0  ...   0.000000   
56039 Teton County, Wyoming      2012  0.000000  0.0  0.0  ...   0.000000   
56041 Uinta County, Wyoming      2012  0.000000  0.0  0.0  ...   0.000000   
56043 Washakie County, Wyoming   2012  0.000000  0.0  0.0  ...   0.000000   
56045 Weston County, Wyoming     2012  0.000000  0.0  0.0  ...   0.000000   

                                                                     \
code                                        624       711       712   
fips  name                       year                                 
1001  Autauga County, Alabama    2012  2.805043  0.000000  0.000000   
1003  Baldwin County, Alabama    2012  7.937621  0.296354  1.046519   
1005  Barbour County, Alabama    2012  0.181091  0.000000  0.000000   
1007  Bibb County, Alabama       2012  0.000000  0.000000  0.000000   
1009  Blount County, Alabama     2012  0.745290  0.000000  0.000000   
...                                         ...       ...       ...   
56037 Sweetwater County, Wyoming 2012  2.144917  0.000000  0.000000   
56039 Teton County, Wyoming      2012  1.220291  1.761712  2.666182   
56041 Uinta County, Wyoming      2012  0.555932  0.00

In [17]:
flattened = pd.DataFrame(table.to_records())
flattened

fips                        name  year  ('ACID', '113')  \
0      1001     Autauga County, Alabama  2012         0.861581   
1      1003     Baldwin County, Alabama  2012         0.533211   
2      1005     Barbour County, Alabama  2012         3.092154   
3      1007        Bibb County, Alabama  2012         1.713780   
4      1009      Blount County, Alabama  2012         0.283806   
...     ...                         ...   ...              ...   
3135  56037  Sweetwater County, Wyoming  2012         0.000000   
3136  56039       Teton County, Wyoming  2012         0.000000   
3137  56041       Uinta County, Wyoming  2012         0.000000   
3138  56043    Washakie County, Wyoming  2012         0.000000   
3139  56045      Weston County, Wyoming  2012         0.000000   

      ('ACID', '114')  ('ACID', '115')  ('ACID', '211')  ('ACID', '212')  \
0                 0.0         0.000000         0.000000         0.000000   
1                 0.0         0.092654         0.000000         2.894175   
2                 0.0         0.000000         0.000000         0.000000   
3                 0.0         0.000000         0.000000         0.000000   
4                 0.0         0.000000         0.000000         0.000000   
...               ...              ...              ...              ...   
3135              0.0         0.172337         0.000000       138.220686   
3136              0.0         0.000000         0.000000         0.000000   
3137              0.0         0.000000         0.000000         0.000000   
3138              0.0         0.000000         5.156900         0.000000   
3139              0.0         0.000000         5.298274         0.000000   

      ('ACID', '213')  ('ACID', '221')  ...  ('WATR', '623')  ('WATR', '624')  \
0            0.000000       284.165607  ...         0.000000         2.805043   
1            0.000000         2.034845  ...        22.062101         7.937621   
2            0.000000         0.000000  ...         0.000000         0.181091   
3            0.000000         0.000000  ...         0.000000         0.000000   
4            0.000000         0.000000  ...         0.715589         0.745290   
...               ...              ...  ...              ...              ...   
3135        52.325460         0.000000  ...         0.000000         2.144917   
3136         0.000000         0.000000  ...         0.000000         1.220291   
3137        27.740218         0.000000  ...         0.000000         0.555932   
3138         0.000000         0.000000  ...         0.000000         3.638850   
3139         2.373792         0.000000  ...         0.000000         0.515444   

      ('WATR', '711')  ('WATR', '712')  ('WATR', '713')  ('WATR', '721')  \
0            0.000000         0.000000       160.007203         6.992977   
1            0.296354         1.046519       545.388688        86.190631   
2            0.000000         0.000000         0.000000         1.582958   
3            0.000000         0.000000         0.000000         0.000000   
4            0.000000         0.000000        17.466768         1.127976   
...               ...              ...              ...              ...   
3135         0.000000         0.000000        41.513813        18.739566   
3136         1.761712         2.666182      1418.001544       187.101821   
3137         0.000000         0.000000         3.241755         7.346062   
3138         0.000000         0.000000         0.000000         0.000000   
3139         0.000000         0.000000         6.531894         0.000000   

      ('WATR', '722')  ('WATR', '811')  ('WATR', '812')  ('WATR', '813')  
0           33.888622         1.546434         4.612651         9.149268  
1          181.539923         9.167564         9.315813        36.941355  
2            8.906419         0.417912         0.442350         1.844123  
3            0.852684         0.571851         0.000000         0.987923  
4           10.630632         1.601868  

In [18]:
# Rewriting column names
r = re.compile('(.*, .*)')
flattened.columns = [fi[-5:-2] + '_' + fi[2:fi.index(", ") - 1] if r.match(fi) is not None else fi 
 for fi in flattened.columns]
flattened

fips                        name  year  113_ACID  114_ACID  115_ACID  \
0      1001     Autauga County, Alabama  2012  0.861581       0.0  0.000000   
1      1003     Baldwin County, Alabama  2012  0.533211       0.0  0.092654   
2      1005     Barbour County, Alabama  2012  3.092154       0.0  0.000000   
3      1007        Bibb County, Alabama  2012  1.713780       0.0  0.000000   
4      1009      Blount County, Alabama  2012  0.283806       0.0  0.000000   
...     ...                         ...   ...       ...       ...       ...   
3135  56037  Sweetwater County, Wyoming  2012  0.000000       0.0  0.172337   
3136  56039       Teton County, Wyoming  2012  0.000000       0.0  0.000000   
3137  56041       Uinta County, Wyoming  2012  0.000000       0.0  0.000000   
3138  56043    Washakie County, Wyoming  2012  0.000000       0.0  0.000000   
3139  56045      Weston County, Wyoming  2012  0.000000       0.0  0.000000   

      211_ACID    212_ACID   213_ACID    221_ACID  ...   623_WATR  624_WATR  \
0     0.000000    0.000000   0.000000  284.165607  ...   0.000000  2.805043   
1     0.000000    2.894175   0.000000    2.034845  ...  22.062101  7.937621   
2     0.000000    0.000000   0.000000    0.000000  ...   0.000000  0.181091   
3     0.000000    0.000000   0.000000    0.000000  ...   0.000000  0.000000   
4     0.000000    0.000000   0.000000    0.000000  ...   0.715589  0.745290   
...        ...         ...        ...         ...  ...        ...       ...   
3135  0.000000  138.220686  52.325460    0.000000  ...   0.000000  2.144917   
3136  0.000000    0.000000   0.000000    0.000000  ...   0.000000  1.220291   
3137  0.000000    0.000000  27.740218    0.000000  ...   0.000000  0.555932   
3138  5.156900    0.000000   0.000000    0.000000  ...   0.000000  3.638850   
3139  5.298274    0.000000   2.373792    0.000000  ...   0.000000  0.515444   

      711_WATR  712_WATR     713_WATR    721_WATR    722_WATR  811_WATR  \
0     0.000000  0.000000   160.007203    6.992977   33.888622  1.546434   
1     0.296354  1.046519   545.388688   86.190631  181.539923  9.167564   
2     0.000000  0.000000     0.000000    1.582958    8.906419  0.417912   
3     0.000000  0.000000     0.000000    0.000000    0.852684  0.571851   
4     0.000000  0.000000    17.466768    1.127976   10.630632  1.601868   
...        ...       ...          ...         ...         ...       ...   
3135  0.000000  0.000000    41.513813   18.739566   32.692556  5.015007   
3136  1.761712  2.666182  1418.001544  187.101821   57.894507  2.318809   
3137  0.000000  0.000000     3.241755    7.346062   10.818613  0.420080   
3138  0.000000  0.000000     0.000000    0.000000    4.126501  0.507885   
3139  0.000000  0.000000     6.531894    0.000000    2.082064  0.317093   

      812_WATR   813_WATR  
0     4.612651   9.149268  
1     9.315813  36.941355  
2     0.442350   1.844123  
3     0.000000   0.987923  
4     0.118085   3.568023  
...        ...        ...  
3135  0.828292   4.295609  
3136  8.456796   9.111469  
3137  0.074580   0.254037  
3138  0.054501   0.516438  
3139  0.000000   0.185040  

[3140 rows x 1683 columns]

In [19]:
#Rearranging the columns so that the input vector is arranged well

flattened = flattened.reindex(sorted(flattened.columns), axis=1)
flattened

113_ACID  113_ENRG     113_ETOX  113_EUTR  113_FOOD     113_GCC  \
0     0.861581  0.275217   893.717406  0.277371  0.818711  141.283089   
1     0.533211  0.170325   553.099157  0.171658  0.506679   87.436540   
2     3.092154  0.987735  3207.488512  0.995466  2.938295  507.055006   
3     1.713780  0.547438  1777.702862  0.551722  1.628506  281.027705   
4     0.283806  0.090657   294.391487  0.091366  0.269684   46.538803   
...        ...       ...          ...       ...       ...         ...   
3135  0.000000  0.000000     0.000000  0.000000  0.000000    0.000000   
3136  0.000000  0.000000     0.000000  0.000000  0.000000    0.000000   
3137  0.000000  0.000000     0.000000  0.000000  0.000000    0.000000   
3138  0.000000  0.000000     0.000000  0.000000  0.000000    0.000000   
3139  0.000000  0.000000     0.000000  0.000000  0.000000    0.000000   

      113_HAPS  113_HAZW        113_HC       113_HNC  ...  813_NREN  \
0     0.120232  0.002792  2.199251e-07  5.000598e-07  ...       0.0   
1     0.074408  0.001728  1.361061e-07  3.094744e-07  ...       0.0   
2     0.431503  0.010021  7.892958e-07  1.794679e-06  ...       0.0   
3     0.239154  0.005554  4.374555e-07  9.946744e-07  ...       0.0   
4     0.039604  0.000920  7.244358e-08  1.647203e-07  ...       0.0   
...        ...       ...           ...           ...  ...       ...   
3135  0.000000  0.000000  0.000000e+00  0.000000e+00  ...       0.0   
3136  0.000000  0.000000  0.000000e+00  0.000000e+00  ...       0.0   
3137  0.000000  0.000000  0.000000e+00  0.000000e+00  ...       0.0   
3138  0.000000  0.000000  0.000000e+00  0.000000e+00  ...       0.0   
3139  0.000000  0.000000  0.000000e+00  0.000000e+00  ...       0.0   

          813_OZON  813_PEST   813_REN  813_SMOG      813_VADD   813_WATR  \
0     9.849320e-10       0.0  0.000129  1.060029   2903.351365   9.149268   
1     3.692511e-09       0.0  0.000484  4.297769  11605.327925  36.941355   
2     0.000000e+00       0.0  0.000000  0.226065    503.263571   1.844123   
3     0.000000e+00       0.0  0.000000  0.121106    269.605484   0.987923   
4     0.000000e+00       0.0  0.000000  0.437393    973.718008   3.568023   
...            ...       ...       ...       ...           ...        ...   
3135  9.117117e-10       0.0  0.000119  0.475895   1741.209694   4.295609   
3136  3.201226e-09       0.0  0.000419  1.026177   7157.274600   9.111469   
3137  0.000000e+00       0.0  0.000000  0.031142     69.327125   0.254037   
3138  0.000000e+00       0.0  0.000000  0.063309    140.936624   0.516438   
3139  0.000000e+00       0.0  0.000000  0.022683     50.497535   0.185040   

       fips                        name  year  
0      1001     Autauga County, Alabama  2012  
1      1003     Baldwin County, Alabama  2012  
2      1005     Barbour County, Alabama  2012  
3      1007        Bibb County, Alabama  2012  
4      1009      Blount County, Alabama  2012  
...     ...                         ...   ...  
3135  56037  Sweetwater County, Wyoming  2012  
3136  56039       Teton County, Wyoming  2012  
3137  56041       Uinta County, Wyoming  2012  
3138  56043    Washakie County, Wyoming  2012  
3139  56045      Weston County, Wyoming  2012  

[3140 rows x 1683 columns]

Hence, each row represents data from a given county. If we look at a row, it coveres data columns in form of [XXX_YYYY] where XXX is the industry code, and YYYY is the USEEIO environmental feature.

In [20]:
flattened.to_csv("modifiedData_US.csv", index=False)